components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2695,10 +2695,27 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return fromDB.value;
     }
 
-    public async getTeams(ctx: TraceContext): Promise<Team[]> {
+    public async getTeams(ctx: TraceContext): Promise<Organization[]> {
         // Note: this operation is per-user only, hence needs no resource guard
-        const user = await this.checkUser("getTeams");
-        return this.teamDB.findTeamsByUser(user.id);
+        const user = await this.checkUser("getOrganizations");
+        const orgs = await this.teamDB.findTeamsByUser(user.id);
+
+        const filterOrg = async (org: Organization): Promise<Organization | undefined> => {
+            const members = await this.teamDB.findMembersByTeam(org.id);
+            if (!(await this.hasOrgOperationPermission(org, members, "get", "read_info"))) {
+                return undefined;
+            }
+            return org;
+        };
+
+        const accessibleOrgs = [];
+        for (const check of orgs.map(filterOrg)) {
+            const org = await check;
+            if (org) {
+                accessibleOrgs.push(org);
+            }
+        }
+        return accessibleOrgs;
```

Comment:
This will make the check run sequentially, not in parallel. Are we happy with that?

Reply:
Or is it the case they are fired in parallel, but just awaited serially?

Reply:
it fires the checks in parallel but then awaits the promises one after the other.

Reply:
Ah, that's neat. Didn't think of doing it that way

Reply:
but doesn't matter much as users don't have many orgs anyway.